packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -641,7 +641,15 @@ export function generateGrid(
                 ${[
                     ...gqlArgs.filter((gqlArg) => gqlArg.queryOrMutationName === gridQueryType.name).map((arg) => arg.name),
                     ...(hasScope ? ["scope"] : []),
-                    ...(hasFilter ? (hasFilterProp ? ["filter: filter ? { and: [gqlFilter, filter] } : gqlFilter"] : ["filter: gqlFilter"]) : []),
+                    ...(filterArg
+                        ? hasFilter && hasFilterProp
+                            ? ["filter: filter ? { and: [gqlFilter, filter] } : gqlFilter"]
```

Comment:
Out of scope: The naming here is confusing: Instead of `gqlFilter` and `filter` I'd use `filter` and `passedFilter`, as both filters are GraphQL filters